vector/src/main/java/im/vector/activity/VectorCallViewActivity.java

```diff
@@ -1279,4 +1289,43 @@ public void onSensorChanged(SensorEvent event) {
     public void onAccuracyChanged(Sensor sensor, int accuracy) {
         Log.d(LOG_TAG, "## onAccuracyChanged(): accuracy=" + accuracy);
     }
+
+    public void endCall() {
+        if (!hasBeenConnectedBeforeEnd && mCall != null && mCall.getIceServers().isEmpty()
+                && !PreferencesManager.useDefaultTurnServer(this)
+                && PreferencesManager.shouldAskForDefaultTurn(this)) {
+            //We should prompt and propose a turn server?
+            String defaultStun = getString(R.string.default_stun_server);
+            if (TextUtils.isEmpty(defaultStun)) {
+                finish();
+                return;
+            }
+
+            AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder
+                    .setTitle(R.string.call_failed_no_ice_title)
+                    .setMessage(
+                            getString(R.string.call_failed_no_ice_description,
+                                    Matrix.getInstance(this).getDefaultSession().getHomeServerConfig().getHomeserverUri(),
+                                    defaultStun)
+                    )
+                    .setPositiveButton(getString(R.string.call_failed_no_ice_use_alt, defaultStun), (dialog, which) -> {
+                        PreferencesManager.setUseDefaultTurnServer(this, true);
+                        MXCallsManager.defaultStunServerUri = defaultStun;
+                        finish();
+                    })
+                    .setNeutralButton(R.string.call_failed_dont_ask_again, (dialog, which) -> {
+                        PreferencesManager.setShouldAskForDefaultTurn(this, false);
+                        finish();
+                    })
+                    .setNegativeButton(R.string.ok, (dialog, which) -> {
```

Comment:
Wording: "OK" may be a bit ambiguous (looking at the screenshot). I think "Cancel" will be better

Reply:
done